ppx/native/ppx_deriving_json_runtime.ml

```diff
@@ -25,9 +40,59 @@ let show_json_type = function
   | `Null -> "null"
   | `String _ -> "string"
 
+let show_json_error ?depth json =
```

Comment:
same comments about truncation of large objects/lists/strings here

Reply:
actually, given that we have `classify` in both browser and native runtimes — can we use same `show_json_error` for both browser and native?

Reply:
Maybe I can attempt it yes. I liked the native code with Format.fprintf a bit better than the browser one with a buffer but I wrote both anyway.

Reply:
This is done.